eva-accession-core/src/test/java/uk/ac/ebi/eva/accession/core/SubmittedVariantAccessioningServiceTest.java

```diff
@@ -104,7 +165,176 @@ public void sameAccessionsAreReturnedForEquivalentVariants() throws AccessionCou
+                                          .collect(Collectors.toList())
+                                          .get(0).getAccession();
+        assertEquals(ACCESSION, accession);
+
+        long accessionDbsnp = submittedVariants.stream()
+                                               .filter(x -> x.getData().getProjectAccession().equals(PROJECT_DBSNP))
+                                               .collect(Collectors.toList())
+                                               .get(0).getAccession();
+        assertEquals(ACCESSION_DBSNP_1, accessionDbsnp);
+
+        assertEquals(3, submittedVariants.size());
+    }
+
+    @UsingDataSet(locations = {"/test-data/submittedVariantEntity.json", "/test-data/dbsnpSubmittedVariantEntity.json"})
+    @Test
+    public void getFromBothRepositories() {
+        List<SubmittedVariant> variants = Arrays.asList(submittedVariant, newSubmittedVariant, dbsnpSubmittedVariant);
+        List<AccessionWrapper<ISubmittedVariant, String, Long>> accessions = service.get(variants);
+        assertEquals(2, accessions.size());
     }
+
+    @UsingDataSet(locations = {"/test-data/submittedVariantEntity.json", "/test-data/dbsnpSubmittedVariantEntity.json"})
+    @Test
+    public void getByAccessionsFromBothRepositories() {
+        List<Long> accessions = Arrays.asList(ACCESSION, ACCESSION_DBSNP_1);
+        List<AccessionWrapper<ISubmittedVariant, String, Long>> submittedVariants = service.getByAccessions(accessions);
+        assertEquals(2, submittedVariants.size());
+    }
+
+    @UsingDataSet(locations = {"/test-data/submittedVariantEntity.json", "/test-data/dbsnpSubmittedVariantEntity.json"})
+    @Test
+    public void getByAccessionAndVersionFromBothRepositories() throws AccessionMergedException,
+            AccessionDoesNotExistException, AccessionDeprecatedException {
+        AccessionWrapper<ISubmittedVariant, String, Long> submittedVariant = service.getByAccessionAndVersion(
+                ACCESSION, 1);
+        assertEquals(this.submittedVariant, submittedVariant.getData());
+
+        AccessionWrapper<ISubmittedVariant, String, Long> dbsnpSubmittedVariant = service.getByAccessionAndVersion(
+                ACCESSION_DBSNP_1, 1);
+        assertEquals(this.dbsnpSubmittedVariant, dbsnpSubmittedVariant.getData());
+    }
+
+    @UsingDataSet(locations = {"/test-data/submittedVariantEntity.json"})
+    @Test
+    public void updateSubmittedVariant() throws AccessionDeprecatedException, AccessionDoesNotExistException,
+            AccessionMergedException, HashAlreadyExistsException {
+        assertUpdate(ACCESSION, submittedVariantModified);
+    }
+
+    private void assertUpdate(long accession, ISubmittedVariant modifiedVariant)
+            throws AccessionDeprecatedException, AccessionDoesNotExistException, AccessionMergedException,
+            HashAlreadyExistsException {
+        service.update(accession, 1, modifiedVariant);
```

Comment:
We are doing the operation to be tested (update) and the assertion in the same method. I think the update should be moved outside this method, and maybe update the method name after that (assertVariantUpdated or something like that)